### PR TITLE
Linac edm2pydm

### PR DESCRIPTION
**Pull request by RafaelLyra8**
- Migration of Linac's BPM Interface from EDM to PyDM
- Insertion of the BPM item on the Launcher Window

image

image

### CODE REVIEW

**Review by murilobalves — APPROVED** [review]

**Review by xresende — APPROVED** [review]

File: pyqt-apps/siriushla/li_di_bpms/bpm_main.py [hunk 1]
```diff
@@ -367,30 +375,30 @@ def display_graph(self):
         return gpVlay
 
     # Get data channel info
-    def dataItem(self, channel, type):
-        if(type == 0):
+    def dataItem(self, channel, style):
+        if(style == 0):
```
Comment (fernandohds564) [inline pyqt-apps/siriushla/li_di_bpms/bpm_main.py:379]: no need for the parenthesis in if statements.

File: pyqt-apps/siriushla/li_di_bpms/bpm_main.py [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+
 import json
 from PyQt5.QtCore import Qt
-from qtpy.QtWidgets import QWidget, QLabel, QGridLayout, QGroupBox, QVBoxLayout, QTabWidget
-from pydm.widgets import PyDMLabel, PyDMLineEdit, enum_button, PyDMEnumComboBox
-from ..widgets import SiriusLedState
+from qtpy.QtWidgets import QGroupBox, QVBoxLayout, QTabWidget
+from qtpy.QtWidgets import QWidget, QLabel, QGridLayout
+from pydm.widgets import PyDMLabel, enum_button, PyDMEnumComboBox, PyDMSpinbox
 from siriushla.as_di_bpms.base import GraphWave
```
Comment (fernandohds564) [inline pyqt-apps/siriushla/li_di_bpms/bpm_main.py:8]: leave an empty line between imports of external libraries and our own.
Also, you can import this class with relative import:
`from ..as_di_bpms.base import GraphWave`.

File: pyqt-apps/scripts/sirius-hla-li-di-bpms.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                    }
+                    QLabel{
+                        padding-right: 5px;
+                    }
+                    GraphWave{
+                        background-color:#000000
+                    }
+                    ''')
+wid.show()
+sys.exit(app.exec_())
```
Comment (fernandohds564) [inline pyqt-apps/scripts/sirius-hla-li-di-bpms.py:45]: need an empty new line at the end of the file.

**Review by fernandohds564 — CHANGES_REQUESTED** [review]

File: pyqt-apps/siriushla/li_di_bpms/bpm_main.py [hunk 4]
```diff
@@ -0,0 +1,483 @@
...
+        return gp_vlay
+
+    def dataItem(self, channel, style):
+        '''Get data channel info'''
+
+        if style == 0:
+            channel_info = PyDMLabel(
+                parent=self,
+                init_channel=self.prefix + self.device_name + ':' + channel)
+        elif(style == 1 or style == 2 or style == 4):
```
Comment (anacso17) [inline pyqt-apps/siriushla/li_di_bpms/bpm_main.py:347]: here you can use 
`elif style in [1, 2, 4]:`

File: pyqt-apps/siriushla/li_di_bpms/bpm_main.py [hunk 5]
```diff
@@ -0,0 +1,483 @@
...
+
+        countx = 0
+        county = 0
+
+        md_glay = QGridLayout()
+
+        for title, info in self.bpm_main_data.items():
+
+            if (title == "Trigger Cnt"
+                  or title == "Cycle"):
```
Comment (anacso17) [inline pyqt-apps/siriushla/li_di_bpms/bpm_main.py:400]: equivalently:
`if title in ["Trigger Cnt", "Cycle"]:`

File: pyqt-apps/scripts/sirius-hla-li-di-bpms.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+import argparse
+import sys
+
+from siriuspy.envars import VACA_PREFIX
+from siriushla.li_di_bpms import DigBeamPosProc
+from siriushla.sirius_application import SiriusApplication
+
+parser = argparse.ArgumentParser(
+    description="Run Interface of Specified Screen.")
```
Comment (anacso17) [inline pyqt-apps/scripts/sirius-hla-li-di-bpms.py:11]: maybe `"Run Interface of Specified BPM."`

File: pyqt-apps/siriushla/li_di_bpms/bpm_main.py [hunk 7]
```diff
@@ -0,0 +1,483 @@
...
+
+class DigBeamPosProc(SiriusMainWindow):
+    ''' Class Digital Beam Position Processor '''
+
+    def __init__(self, device_name, prefix='', parent=None):
+        '''Contain all the graphic interface data'''
+
+        super().__init__(parent)
+
+        self.prefix = prefix
```
Comment (anacso17) [inline pyqt-apps/siriushla/li_di_bpms/bpm_main.py:21]: I think this line is not necessary

File: pyqt-apps/siriushla/li_di_bpms/bpm_main.py [hunk 8]
```diff
@@ -0,0 +1,482 @@
...
+        self.selectors_data = {
+            "Trigger Mode": "ACQ_TRIGGER",
+            "Position Algorithm": "POS_ALG"
+            }
+
+        self._setupUi()
+
+    def _setupUi(self):
+        '''Build the graphic interface'''
+
```
Comment (fernandohds564) [inline pyqt-apps/siriushla/li_di_bpms/bpm_main.py:241]: there is no empty line after the docsctring of a method

File: pyqt-apps/siriushla/li_di_bpms/bpm_main.py [hunk 9]
```diff
@@ -0,0 +1,482 @@
...
+        '''Display all main data widgets'''
+
+        countx = 0
+        county = 0
+
+        md_glay = QGridLayout()
+
+        for title, info in self.bpm_main_data.items():
+
+            if (title == "Trigger Cnt"
```
Comment (fernandohds564) [inline pyqt-apps/siriushla/li_di_bpms/bpm_main.py:398]: do you really need this line break here? it seems the code would fit in a single line.

File: pyqt-apps/siriushla/li_di_bpms/bpm_main.py [hunk 10]
```diff
@@ -0,0 +1,482 @@
...
+        return gp_vlay
+
+    def dataItem(self, channel, style):
+        '''Get data channel info'''
+
+        if style == 0:
+            channel_info = PyDMLabel(
+                parent=self,
+                init_channel=self.prefix + self.device_name + ':' + channel)
+        elif(style == 1 or style == 2 or style == 4):
```
Comment (fernandohds564) [inline pyqt-apps/siriushla/li_di_bpms/bpm_main.py:346]: no need for parenthesis in a if clause.

**Review by fernandohds564 — CHANGES_REQUESTED** [review]
very nice @RafaelLyra8 ! Just a few more changes and we are good to merge!

**Review by fernandohds564 — APPROVED** [review]
very nice PR! Congrats!

**Pull request merged**